ray-operator/controllers/ray/rayservice_controller.go

```diff
@@ -295,8 +295,9 @@ func (r *RayServiceReconciler) calculateStatus(ctx context.Context, rayServiceIn
 				"newCluster", clusterName)
 			rayServiceInstance.Status.ActiveServiceStatus = rayServiceInstance.Status.PendingServiceStatus
 			rayServiceInstance.Status.PendingServiceStatus = rayv1.RayServiceStatus{}
-			rayServiceInstance.Status.ServiceStatus = rayv1.Running
 		}
+		// make ServiceStatus ready since headSvc and serveSvc are ready.
+		rayServiceInstance.Status.ServiceStatus = rayv1.Running
```

Comment:
We should mark `ServiceStatus = rayv1.Running` not only when promoting but also when headSvc and serveSvc are ready; otherwise, the test won't pass.

Reply:
`Running` should have the same definition as `Ready` condition.

Reply:
maybe you can open another follow up PR to address it.